Library/Homebrew/test/support/helper/env.rb

```diff
@@ -1,13 +1,31 @@
 module Test
   module Helper
     module Env
+      def copy_env
+        ENV.to_hash
+      end
+
+      def restore_env(env)
+        ENV.replace(env)
+      end
+
       def with_environment(partial_env)
-        old = ENV.to_hash
+        old = copy_env
         ENV.update partial_env
-        begin
+        yield
+      ensure
+        restore_env old
+      end
+
+      def using_git_env
```

Comment:
Trivia I just discovered: the previous value was the timestamp of the initial commit to Homebrew.